### PR TITLE
add fabric api message

### PR DESCRIPTION
**Pull request by shedaniel**
since this message is actually already on fabricmc.net/use, I don't think this is bad to add it.

I am actually sceptical if HTML should be in the language bundle, but I don't see many ways around it that will be neat.

image

### CODE REVIEW

File: src/main/java/net/fabricmc/installer/client/ClientHandler.java [hunk 1]
```diff
@@ -52,13 +57,29 @@ public void install() {
...
 		}).start();
 	}
 
+	private void showInstalledMessage(String loaderVersion, String gameVersion) {
+		JEditorPane pane = new JEditorPane("text/html", new MessageFormat(Utils.BUNDLE.getString("prompt.install.successful")).format(new Object[]{loaderVersion, gameVersion, Reference.fabricApiUrl}));
+		pane.setEditable(false);
+		pane.addHyperlinkListener(e -> {
+			if (e.getEventType() == HyperlinkEvent.EventType.ACTIVATED) {
+				try {
+					Desktop.getDesktop().browse(e.getURL().toURI());
```
Comment (UpcraftLP) [inline src/main/java/net/fabricmc/installer/client/ClientHandler.java:74]: pretty sure this will crash on MacOS

Comment (modmuss50) [reply src/main/java/net/fabricmc/installer/client/ClientHandler.java:74]: This needs testing, not sure why it would though.

File: src/main/resources/lang/installer.properties [hunk 2]
```diff
@@ -30,5 +30,7 @@ prompt.server.jar.valid=Valid {0} server jar found
 prompt.server.jar.invalid=No valid {0} server jar found
 prompt.server.generate=Generate
 prompt.server.overwrite=Are you sure you want to override the existing launch scripts?
+prompt.install.successful.title=Successfully Installed
+prompt.install.successful=Fabric Loader {0} for {1} has been successfully installed.<br>Most mods will also require you to install <a href="{2}">Fabric (API)</a>.
```
Comment (Juuxel) [inline src/main/resources/lang/installer.properties:34]:
```suggestion
prompt.install.successful=Fabric Loader {0} for {1} has been successfully installed.<br>Most mods will also require you to install <a href="{2}">Fabric API</a>.
```

Comment (shedaniel) [reply src/main/resources/lang/installer.properties:34]: It is called Fabric (API) in the use page and fabric api CF page as well, I think this is just to bring uniformity, athrough I prefer Fabric API as well

**Comment by shedaniel** [comment]
i've changed the error message into JEditorPane to allow selecting the text

File: src/main/java/net/fabricmc/installer/client/ClientHandler.java [hunk 3]
```diff
@@ -52,13 +57,33 @@ public void install() {
...
 			}
 			buttonInstall.setEnabled(true);
 		}).start();
 	}
 
+	private void showInstalledMessage(String loaderVersion, String gameVersion) {
+		JEditorPane pane = new JEditorPane("text/html", new MessageFormat(Utils.BUNDLE.getString("prompt.install.successful")).format(new Object[]{loaderVersion, gameVersion, Reference.fabricApiUrl}));
+		pane.setEditable(false);
+		pane.addHyperlinkListener(e -> {
+			try {
```
Comment (hYdos) [inline src/main/java/net/fabricmc/installer/client/ClientHandler.java:72]: weird spacing on this line. could conflict with style of code

Comment (liach) [reply src/main/java/net/fabricmc/installer/client/ClientHandler.java:72]: This is just one tab indent. Look at the line above, this is a new code block so it needs to indent

File: src/main/java/net/fabricmc/installer/client/ClientHandler.java [hunk 4]
```diff
@@ -52,13 +57,33 @@ public void install() {
...
+				if (e.getEventType() == HyperlinkEvent.EventType.ACTIVATED) {
+					if (Desktop.isDesktopSupported() && Desktop.getDesktop().isSupported(Desktop.Action.BROWSE)) {
+						Desktop.getDesktop().browse(e.getURL().toURI());
+					} else {
+						throw new UnsupportedOperationException("Failed to open " + e.getURL().toString());
+					}
+				}
+			} catch (Exception exception) {
```
Comment (liach) [inline src/main/java/net/fabricmc/installer/client/ClientHandler.java:80]: Maybe just catch any throwable

Comment (shedaniel) [reply src/main/java/net/fabricmc/installer/client/ClientHandler.java:80]: sure

File: src/main/java/net/fabricmc/installer/client/ClientHandler.java (hunk 1 above)
Comment (haykam821) [reply src/main/java/net/fabricmc/installer/client/ClientHandler.java]: I didn't crash on macOS, but the font and background color doesn't match the rest of the box:

macOS Fabric API install popup

Comment (shedaniel) [reply src/main/java/net/fabricmc/installer/client/ClientHandler.java]: Oh yeah I have to set the font, give me a second.

Comment (haykam821) [reply src/main/java/net/fabricmc/installer/client/ClientHandler.java]: The formatting issues have been fixed in 9359e301a7e53ead97055cf02ed5f5545e141dad, so now macOS support is complete:

macOS Fabric API install popup with correct formatting

File: src/main/resources/lang/installer.properties (hunk 2 above)
Comment (Juuxel) [reply src/main/resources/lang/installer.properties]: API on cf
It's been called Fabric API without parentheses on CF as well for some time now.

Comment (shedaniel) [reply src/main/resources/lang/installer.properties]: Don't worry, I've renamed this to use without parentheses already now :)

File: src/main/java/net/fabricmc/installer/Handler.java [hunk 5]
```diff
@@ -110,12 +109,12 @@ public JPanel makePanel(InstallerGui installerGui) {
 			for (int i = 0; i < versions.size(); i++) {
 				MetaHandler.GameVersion version = versions.get(i);
 				loaderVersionComboBox.addItem(version.getVersion());
-				if(version.isStable()){
+				if (version.isStable()) {
```
Comment (UpcraftLP) [inline src/main/java/net/fabricmc/installer/Handler.java:112]: unnecessary change

File: src/main/java/net/fabricmc/installer/Handler.java [hunk 6]
```diff
@@ -110,12 +109,12 @@ public JPanel makePanel(InstallerGui installerGui) {
...
 					stableIndex = i;
 				}
 			}
 			//If no stable versions are found, default to the latest version
-			if(stableIndex == -1){
+			if (stableIndex == -1) {
```
Comment (UpcraftLP) [inline src/main/java/net/fabricmc/installer/Handler.java:117]: unnecessary change

File: src/main/java/net/fabricmc/installer/client/ClientHandler.java [hunk 7]
```diff
@@ -70,7 +100,7 @@ public void installCli(ArgumentParser args) throws Exception {
 		String loaderVersion = getLoaderVersion(args);
 
 		String profileName = ClientInstaller.install(file, gameVersion, loaderVersion, InstallerProgress.CONSOLE);
-		if(args.has("noprofile")) {
+		if (args.has("noprofile")) {
```
Comment (UpcraftLP) [inline src/main/java/net/fabricmc/installer/client/ClientHandler.java:103]: unnecessary change

**Comment by shedaniel** [comment]
Removed unnecessary change

**Comment by liach** [comment]
rip

**Comment by TheGlitch76** [comment]
>  it should be part of the "Fabric Installed" message in my opinion
https://cdn.discordapp.com/attachments/434705643388272672/713444222560108564/unknown.png
